Core/LinkCleaner.swift

```diff
@@ -88,4 +90,38 @@ public class LinkCleaner {
         
         return nil
     }
+    
+    public func cleanTrackingParameters(initiator: URL?, url: URL?,
+                                        config: PrivacyConfiguration = PrivacyConfigurationManager.shared.privacyConfig) -> URL? {
+        urlParametersRemoved = false
+        guard config.isEnabled(featureKey: .trackingParameters) else { return url }
+        guard let url = url, !isURLExcluded(url: url, config: config, feature: .trackingParameters) else { return url }
+        if let initiator = initiator, isURLExcluded(url: initiator, config: config, feature: .trackingParameters) {
+            return url
+        }
+        
+        guard var urlsComps = URLComponents(url: url, resolvingAgainstBaseURL: false) else {
+            return url
+        }
+        guard let queryParams = urlsComps.queryItems, queryParams.count > 0 else {
+            return url
+        }
+        
+        let trackingParams = TrackingLinkSettings(fromConfig: config).trackingParameters
+        
+        let preservedParams: [URLQueryItem] = queryParams.filter { param in
+            for trackingParam in trackingParams {
+                if param.name.matches(pattern: "^\(trackingParam)$") {
+                    urlParametersRemoved = true
+                    return false
+                }
+            }
```

Comment:
Tiny Swift syntax note, not a blocker: we can use `for x in y where z` syntax to reduce some nesting:

```suggestion
            for trackingParam in trackingParams where param.name.matches(pattern: "^\(trackingParam)$") {
                urlParametersRemoved = true
                return false
            }
```

That said, since we're returning on the first match what we're really doing here is equivalent to `contains`:

```suggestion
            if trackingParams.contains(where: { param.name.matches(pattern: "^\($0)$") }) {
                urlParametersRemoved = true
                return false
            }
```